### PR TITLE
bluetooth: rpc: Clean up

### PR DESCRIPTION
**Pull request by KAGA164**
This is basic clean-up for the Bluetooth RPC, it contains:

- Improved string serializers
- Test cases for Bluetooth samples that can be build out of the box with serialized BLE stack
- Fix for the unused bt_conn_pairing_feat_buf_size variable

### CODE REVIEW

File: samples/bluetooth/peripheral_hids_mouse/sample.yaml [hunk 1]
```diff
@@ -21,3 +21,11 @@ tests:
     platform_allow: nrf51dk_nrf51422 nrf52dk_nrf52832 nrf52840dk_nrf52840 nrf5340dk_nrf5340_cpuapp
       nrf5340dk_nrf5340_cpuapp_ns
     tags: bluetooth ci_build
+  sample.bluetooth.peripheral_hids_mouse.ble_rpc:
+    build_only: true
+    extra_configs:
+      - CONFIG_BT_RPC_STACK=y
+    integration_platforms:
+      - nrf5340dk_nrf5340_cpuapp
```
Comment (kapi-no) [inline samples/bluetooth/peripheral_hids_mouse/sample.yaml:29]: is it possible to run nRF RPC with the ns target? If so, we can also add NS board variants to the test yaml

**Review by kapi-no — APPROVED** [review]

File: samples/bluetooth/peripheral_hids_mouse/sample.yaml (hunk 1 above)
Comment (KAGA164) [reply samples/bluetooth/peripheral_hids_mouse/sample.yaml:29]: At this moment It cannot be built for ns target. Seems like there are some additional dependencies that would need to be serialized to run this on non secure target

File: subsys/bluetooth/rpc/common/serialize.c [hunk 2]
```diff
@@ -324,6 +324,51 @@ int32_t ser_decode_int(CborValue *value)
 	return 0;
 }
 
+void ser_decode_str(CborValue *value, char *buffer, size_t buffer_size)
```
Comment (de-nordic) [inline subsys/bluetooth/rpc/common/serialize.c:327]: Most of the `ser_` decoding functions that process buffers return buffer pointer or NULL when they fail, this allows quickly to find out whether decoding went fine. Can we do that here too?

**Review by doki-nordic — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/rpc/common/serialize.c [hunk 3]
```diff
@@ -428,6 +473,9 @@ char *ser_decode_str_into_scratchpad(struct ser_scratchpad *scratchpad)
 			goto error_exit;
 		}
 
+		/* Reserve place for a string NULL terminator. */
+		len++;
```
[the remark this answers is not in the document]
Comment (KAGA164) [reply subsys/bluetooth/rpc/common/serialize.c:477]: This function is used here to decode a string into scratchpad and it doesn't work correctly without NULL terminator. There is different function used for buffer decoding into scratchpad

File: subsys/bluetooth/rpc/common/serialize.c (hunk 2 above)
Comment (KAGA164) [reply subsys/bluetooth/rpc/common/serialize.c:327]: Sure will be done under #7655

**Comment by KAGA164** [comment]
Will be done as part of the https://github.com/nrfconnect/sdk-nrf/pull/7655

**Pull request closed without merging**